packages/graph/hash_graph/lib/graph/src/types/schema/data.rs

```diff
@@ -0,0 +1,316 @@
+use std::collections::HashMap;
+
+use serde::{Deserialize, Serialize};
+use serde_json::json;
+
+use crate::types::schema::{Uri, Validate, ValidationError};
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(
+    try_from = "DataTypeRepr",
+    tag = "kind",
+    rename = "dataType",
+    rename_all = "camelCase"
+)]
+pub struct DataType {
+    #[serde(rename = "$id")]
+    id: Uri,
+    title: String,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    description: Option<String>,
+    #[serde(rename = "type")]
+    ty: String,
+    #[serde(flatten)]
+    additional_properties: HashMap<String, serde_json::Value>,
```

Comment:
Perhaps a comment saying that this is a catch-all for now until we better define non-primitive DataTypes.

Reply:
Added a comment in 61792dd64a377f0bf04f3d65492e22c19cb00ca0